### PR TITLE
Fix/improve Maestro DB migration

### PR DESCRIPTION
**Pull request by premun**
We no longer pull a `.nupkg` from `nuget.org` directly but restore the `EF` tool correctly from the config and call the tool instead of calling into a DLL inside of it.

Fixing the build after we moved to 8.0 in https://github.com/dotnet/arcade-services/issues/3053

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Improved Maestro DB migration

### CODE REVIEW

File: test/Directory.Build.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project>
   <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
   <PropertyGroup>
-    <TargetFramework>net8.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (oleksandr-didyk) [inline test/Directory.Build.props:5]: Why do we need to downgrade the TFM here?

**Review by oleksandr-didyk — APPROVED** [review]

File: test/Directory.Build.props (hunk 1 above)
Comment (premun) [reply test/Directory.Build.props:5]: The `src/` projects are still `net6.0` but the tests got bumped to `8.0` which was causing some problems

Comment (dkurepa) [inline test/Directory.Build.props:5]: why is this set to 6.0? Aren't we moving to 8.0?

Comment (premun) [reply test/Directory.Build.props:5]: The `src/` projects are still `net6.0` but the tests got bumped to `8.0` which was causing some problems. We actually are not moving the projects much, just Arcade

**Pull request merged**